### PR TITLE
Refactor `AutoCompleteStringCollection` to use `List<string>` instead of `ArrayList`

### PR DESCRIPTION
**Pull request by elachlan**
Work related to #8140

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8142)

### CODE REVIEW

**Comment by elachlan** [comment]
@RussKie flaky test? I can't see any test failures.

**Comment by RussKie** [comment]
This looks like a new regression introduced by https://github.com/dotnet/winforms/pull/8105.

**Comment by elachlan** [comment]
Could you please rerun the pipeline?

**Comment by RussKie** [comment]
> Could you please rerun the pipeline?

Done.
You could restart the build pipeline by closing the PR and then reopening it.

File: src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace System.Windows.Forms
     public class AutoCompleteStringCollection : IList
     {
         CollectionChangeEventHandler onCollectionChanged;
-        private readonly ArrayList data = new ArrayList();
+        private readonly List<string> data = new List<string>();
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:18]:
```suggestion
        private readonly List<string> data = new();
```

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs [hunk 2]
```diff
@@ -42,48 +42,28 @@ public AutoCompleteStringCollection()
...
 
         /// <summary>
         ///  Adds a string with the specified value to the
         ///  <see cref="AutoCompleteStringCollection"/> .
         /// </summary>
         public int Add(string value)
         {
-            int index = data.Add(value);
+            data.Add(value);
+            int index = data.LastIndexOf(value);
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:66]: Curious, is this faster than calling `data.Count - 1`?

**Review by lonitra — APPROVED** [review]
Just a question. Otherwise LGTM, thank you!

**Review by lonitra — APPROVED** [review]

**Pull request merged**